### PR TITLE
docs: Add to the schema

### PR DESCRIPTION
**Pull request by ieQu1**

### CODE REVIEW

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_conf/src/emqx_conf_schema.erl [hunk 1]
```diff
@@ -112,135 +112,173 @@ fields("cluster") ->
...
     [ {"addr",
        sc(string(),
           #{ default => "239.192.0.1"
+           , desc => "Multicast IPv4 address."
            })}
     , {"ports",
        sc(hoconsc:array(integer()),
           #{ default => [4369, 4370]
+           , desc => "List of UDP ports used for service discovery.<br/>
+Note: probe messages are broadcasted to all the specified ports."
```
[the remark this answers is not in the document]
Comment (ieQu1) [reply apps/emqx_conf/src/emqx_conf_schema.erl]: Fixed.

**Review by thalesmg — APPROVED** [review]

**Comment by zmstone** [comment]
we can do this too:
```
fileds(StructRef) ->
    #{ fields => [ .... ],
     , desc => "struct description"
     }
```

**Review by zmstone — APPROVED** [review]

File: apps/emqx/src/emqx_schema.erl [hunk 2]
```diff
@@ -1079,65 +1158,114 @@ fields("deflate_opts") ->
...
         {"session_locking_strategy",
             sc(
                 hoconsc:enum([local, leader, quorum, all]),
-                #{default => quorum}
+                #{
+                    default => quorum,
+                    desc =>
+                        "Session locking strategy in a cluster.<br/>\n"
+                        " - `local`: only lock the session on the current node\n"
+                        " - `one`: select only one remove node to lock the session\n"
```
Comment (thalesmg) [inline apps/emqx/src/emqx_schema.erl:1231]:
```suggestion
                        " - `one`: select only one remote node to lock the session\n"
```
?

File: apps/emqx/src/emqx_schema.erl [hunk 3]
```diff
@@ -1079,65 +1158,114 @@ fields("deflate_opts") ->
...
                 hoconsc:enum([random, round_robin, sticky, hash_topic, hash_clientid]),
-                #{default => round_robin}
+                #{
+                    default => round_robin,
+                    desc =>
+                        "Dispatch strategy for shared subscription.<br/>\n"
+                        " - `random`: dispatch the message to a random selected subscriber\n"
+                        " - `round_robin`: select the subscribers in a round-robin manner\n"
+                        " - `sticky`: always use the last selected subscriber to dispatch,\n"
+                        "   until the subscriber disconnected.\n"
```
Comment (thalesmg) [inline apps/emqx/src/emqx_schema.erl:1246]:
```suggestion
                        "   until the subscriber disconnects.\n"
```

File: apps/emqx/src/emqx_schema.erl [hunk 4]
```diff
@@ -1079,65 +1158,114 @@ fields("deflate_opts") ->
...
             sc(
                 boolean(),
-                #{default => false}
+                #{
+                    default => false,
+                    desc =>
+                        "Enable/disable shared dispatch acknowledgement for QoS1 and QoS2 messages.<br/>\n"
+                        " This should allow messages to be dispatched to a different subscriber in\n"
+                        " the group in case the picked (based on `shared_subscription_strategy`) subscriber\n"
+                        " is offline."
```
Comment (thalesmg) [inline apps/emqx/src/emqx_schema.erl:1259]: For my own information: this re-routing to an online subscriber can happen when this parameter is enabled?

File: apps/emqx/src/emqx_schema.erl [hunk 5]
```diff
@@ -1248,75 +1376,134 @@ fields("sysmon_vm") ->
...
+                        " https://github.com/erlang/otp/blob/feb45017da36be78d4c5784d758ede619fa7bfd3/erts/emulator/beam/erl_gc.c#L421"
+                }
             )},
         {"long_schedule",
             sc(
                 hoconsc:union([disabled, duration()]),
-                #{default => "240ms"}
+                #{
+                    default => "240ms",
+                    desc => "Enable Long Schedule(ms) monitoring."
```
Comment (thalesmg) [inline apps/emqx/src/emqx_schema.erl:1421]:
```suggestion
                    desc => "Enable Long Schedule monitoring."
```
?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx/src/emqx_schema.erl (hunk 4 above)
Comment (ieQu1) [reply apps/emqx/src/emqx_schema.erl:1259]: Not sure about the correct answer: I copied that from `emqx.conf.all`

**Review by savonarola — APPROVED** [review]

File: apps/emqx/src/emqx_schema.erl [hunk 6]
```diff
@@ -1488,27 +1675,57 @@ base_listener() ->
...
+                    desc => "IP address and port for the listening socket."
+                }
             )},
         {"acceptors",
             sc(
                 integer(),
-                #{default => 16}
+                #{
+                    default => 16,
+                    desc => "The size of the listener's receiving pool."
```
Comment (qzhuyan) [inline apps/emqx/src/emqx_schema.erl:1688]: nit, acceptors pool.

I guess it is copied from somewhere?

**Review by qzhuyan — APPROVED** [review]

File: apps/emqx/src/emqx_schema.erl (hunk 6 above)
Comment (ieQu1) [reply apps/emqx/src/emqx_schema.erl:1688]: Yes, indeed. Will fix in the next PR.

**Pull request merged**